src/main/scala/io/iohk/ethereum/blockchain/sync/PeersClient.scala

```diff
@@ -129,6 +136,25 @@ class PeersClient(
 
     log.debug(s"Handshaked peers status (number of peers: ${handshakedPeersStatus.size}): $handshakedPeersStatus")
   }
+
+  //returns the next best peer after the one already returned previously
+  //TODO: make sure the next best peer has a different best block, so we don't fetch in parallel identical branches
+  private def nextBestPeer(peersToDownloadFrom: Map[PeerId, PeerWithInfo]): Option[Peer] = {
```

Comment:
this could (and IMO should) be a public function in an object. and it should be covered by a set of unit tests to document the expectation (`activeFetchingNodes.contains(peerAndWeight._1)).maxByOption(_._2).map(_._1)` in not obvious at first glance)